backend/src/apiserver/client_manager.go

```diff
@@ -241,6 +246,11 @@ func initDBClient(initConnectionTimeout time.Duration) *storage.DB {
 		glog.Fatalf("Failed to create index experimentuuid_conditions_finishedatinsec on run_details. Error: %s", response.Error)
 	}
 
+	response = db.Model(&model.Pipeline{}).AddUniqueIndex("name_namespace_index", "Name", "Namespace")
```

Comment:
>> We removed the unique here for Name but we should make the tuple(Namespace, Name) unique.

@capri-xiyue  The creation of a unique index with (Namespace, Name) is done in the client manager. The reason we can't do this as part of the Pipeline struct definition is because the KFP repository is using a branch of the gorm package ```"github.com/jinzhu/gorm"``` which does not support it, the refactoring for this is not in scope.

Reply:
Thank you, created issue https://github.com/kubeflow/pipelines/issues/5125 to track it